include/zephyr/net/conn_mgr_connectivity_impl.h

```diff
@@ -252,6 +252,39 @@ static inline struct conn_mgr_conn_binding *conn_mgr_if_get_binding(struct net_i
 	return NULL;
 }
 
+/**
+ * @brief Claims the passed-in binding, making it save to access.
+ *
+ * Call this whenever accessing binding data, unless inside a conn_mgr_conn_api callback, where it
+ * is called automatically by conn_mgr.
+ *
+ * Reentrant.
+ *
+ * For use only by connectivity implementations.
+ *
+ * @param binding - Binding to claim
+ */
+static inline void conn_mgr_binding_take(struct conn_mgr_conn_binding *binding)
```

Comment:
Could be me, but I'd find `conn_mgr_binding_lock()/unlock()` a bit more intuitive.